docs/multicluster/architecture.md

```diff
@@ -0,0 +1,95 @@
+# Antrea Multi-cluster Architecture
+
+Antrea Multi-cluster implements [Multi-cluster Service API](https://github.com/kubernetes/enhancements/tree/master/keps/sig-multicluster/1645-multi-cluster-services-api) for Service communication
+across multiple Kubernetes clusters.
+
+Below diagram depicts a basic multi-cluster topology in Antrea.
+
+<img src="assets/basic-topology.png" width="400" alt="Antrea Multi-cluster Topology">
+
+Given a set of Kubernetes clusters, there will be a leader cluster and several member clusters.
+By default, a leader cluster itself is also a member cluster of a ClusterSet. A cluster
+can also be configured as a dedicated leader cluster of multiple ClusterSets.
+
+## Terminology
+
+ClusterSet is a placeholder name for a group of clusters with a high degree of mutual
+trust and shared ownership that share Services amongst themselves. Within a ClusterSet,
+Namespace sameness applies, which means all Namespaces with a given name are considered to
+be the same Namespace. It's also a Custom Resource Definition which defines a ClusterSet
+including the leader and member clusters information.
+
+ClusterClaim is a Custom Resource Definition which can be used to define a cluster itself as
+a member of a ClusterSet with unique cluster ID, or claim a ClusterSet with unique ClusterSet
+ID.
+
+MemberClusterAnnounce is a Custom Resource Definition which declares a member cluster
+configuration to the leader cluster.
+
+Common Area is an abstraction in Multi-cluster implementation that provides interfaces to a
```

Comment:
the Antrea Multi-cluster implementation

Reply:
How about "provides a storage interface for resource export/import that can be read/written by all member and leader clusters in the ClusterSet"?

We should add "The Common Area is implemented with a Namespace in the leader cluster".